src/Print.jsx

```diff
@@ -13,76 +12,61 @@ const propTypes = {
   exclusive: PropTypes.bool,
   printOnly: PropTypes.bool
 };
-const contextTypes = PrintProvider.childContextTypes;
-
-export default class Print extends React.Component {
-
-  constructor(props) {
-    super(props);
 
-    this.state = {
-      printOffsetLeft: 0,
-      printOffsetTop: 0
-    };
-  }
-  componentDidMount() {
+export default function Print (props) {
+  const [printOffsetLeft, setPrintOffsetLeft] = useState(0);
+  const [printOffsetTop, setPrintOffsetTop] = useState(0);
+  const { regPrintable, unregPrintable } = usePrintProvider();
+  const printElement = useRef(null);
+  const main_ = (props.main || props.single) ? s._main : '';
+  const excl_ = (props.exclusive || props.printOnly) ? s._exclusive : '';
+  const isPrint = window.matchMedia('print').matches;
+  const offset_ = ((printOffsetTop || printOffsetLeft) && main_ && isPrint) ?
+    { marginTop: -printOffsetTop, marginLeft: -printOffsetLeft } : {};
+  const globalClassName = 'react-easy-print-print'; // using in hiddenAll
+  const className = `${globalClassName} ${s.root} ${main_} ${excl_}`;
 
-    if (this.props.name) {
-      debug('init printable', this.props.name);
-      const isSingle = (this.props.main || this.props.single);
-      this.context.printProvider && this.context.printProvider.regPrintable(this.props.name, <Print {...this.props} />, isSingle);
+  useEffect(() => {
```

Comment:
looks like this useEffect should be componentDidMount equivalent, because, e.g. window.matchMedia('print').onchange must be called at once,

so, as I understand, we need add empty  list of dependencies (`[]`) here: `useEffect(() => {...}, []);`

Reply:
also, we need verify all the useEffects' dependencies